### ISSUE TITLE
Add documentation about GitHub token needed for Branch Protection

### ISSUE THREAD

**Issue by laurentsimon** [ossf/scorecard#1106]
Admin tokens give extra information, so we need to document that

**Comment by laurentsimon** [comment]
FYI  @olivekl

**Comment by olivekl** [comment]
I had included the following to the updates I'm working on: 

"Requirements: Authorization token must have Admin access to repository"

"Without an authorization token with Admin access to the repository, the check will return a failing score."

Is there anything else you'd like included?

**Comment by laurentsimon** [comment]
I think @azeemsgoogle has more details to add. The check won't fail but some settings may not be reported correctly. We need to document which settings these are, I think.

**Comment by naveensrinivasan** [comment]
We don't want to recommend using admin tokens.

**Comment by naveensrinivasan** [comment]
> How about something along the lines of:
> 
> ```
> Some settings queried by the Branch-Protection check require an admin token. 
> If the provided token does not have admin access, the check will query branch settings 
> accessible to non-admins and provide results based on these settings. We recommend 
> using a non-admin token since this provides a thorough enough result needed by most users.  
> ```

Sounds good!

**Comment by laurentsimon** [comment]
Can we call out which settings are not available without admin token? That's really useful info for anyone who wants to understand the results.

**Comment by azeemshaikh38** [comment]
> I'm going to think about it more as part os the local directory repo client, since I'm working on it.

Yes, if some of these APIs can be made more generic, that'll be great. Thanks for looking into this.

> For now, since we only support GitHub, it's useful for users to know that the setting you listed may give erroneous values if they don't provide an admin token.

If you feel strongly that exposing this info helps users (and does not confuse them with too much details), then sure, let's document them. Maybe avoid the word `erroneous` though.

> We could also decide to remove these settings from the check calculation altogether since they cannot be used for score calculation in the common case? This would go in the direction of #1027

I would prefer that we leave these checks as is for now. It may be true that users shouldn't require them, but that does not mean Scorecard shouldn't be reporting on them.
